packages/docs/scripts/generate-docs.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Requires jsdoc and jsdoc-to-markdown
+ * Reference: https://gist.github.com/slorber/0bf8c8c8001505f0f99a062ac55bf442
+ */
+
+/* eslint-disable no-console */
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const jsdoc2md = require('jsdoc-to-markdown');
+const { execSync } = require('child_process');
+const sidebars = require('../sidebars');
+
+/**
+ * Runs through packages folders looking for JSDoc and generates markdown docs
+ */
+function generateDocs() {
+    console.log('-= Generating package docs =-');
+    // Use glob to get all js/ts files
+    const pathPattern = path.join(
+        __dirname,
+        '../../core/src/**/*.[jt]s?(x)'
+        // '../packages/**/src/**/*.[jt]s?(x)'
+    );
+    const filePaths = glob.sync(pathPattern, {
+        ignore: [
+            '**/node_modules/**',
+            '**/*.spec.js',
+            '**/*.d.ts',
+            '**/index.js',
+            '**/*.stories.js',
+            '**/*DataBuilder.js',
+            '**/*Data.js',
+        ],
+    });
+
+    // Get the sidebar object
+    let processingPackageName;
+
+    // grab all js files
+    filePaths.forEach((filePath) => {
+        // Generate markdown from JSDoc comments
+        const markdown = jsdoc2md.renderSync({
+            files: filePath,
+            configure: path.join(__dirname, '../jsdoc.conf.json'),
+            // plugin: path.join(__dirname, './template.hbs'),
+            'heading-depth': 1,
+        });
+
+        // if there's markdown, do stuff
+        if (markdown && markdown.length > 0) {
+            // get the package ID from the file path
+            const matchingExpression = filePath.match(
+                /\/packages\/([\s\S]*?)\/src\/charts\/([\s\S]*?)\//i
```

Comment:
could you have it as a const and link to an example that will match the path? just for future self is always nice to find that around a regex :)

Reply:
Great idea